OutOfSchool/OutOfSchool.WebApi/Services/StatisticService.cs

```diff
@@ -57,115 +62,141 @@ public async Task<IEnumerable<DirectionStatistic>> GetPopularDirections(int limi
 
-            // LEFT JOIN DirectionsWithWorkshops with DirectionsWithApplications
-            var directionsWithCounts = directionsWithWorkshops
-                .GroupJoin(
-                    directionsWithApplications,
-                    directionWithWorkshop => directionWithWorkshop.DirectionId,
-                    directionWithApplication => directionWithApplication.DirectionId,
-                    (directionWithWorkshop, localDirectionsWithApplications) => new
-                    {
-                        directionWithWorkshop,
-                        localDirectionsWithApplications,
-                    })
-                .SelectMany(
-                    x => x.localDirectionsWithApplications.DefaultIfEmpty(),
-                    (x, y) => new
+                var directionsWithWorkshops = workshops
+                    .GroupBy(w => w.DirectionId)
+                    .Select(g => new
                     {
-                        DirectionId = x.directionWithWorkshop.DirectionId,
-                        ApplicationsCount = y.ApplicationsCount,
-                        WorkshopsCount = x.directionWithWorkshop.WorkshopsCount,
+                        DirectionId = g.Key,
+                        WorkshopsCount = g.Count() as int?,
                     });
 
-            var allDirections = directionRepository.Get<int>();
-
-            // LEFT JOIN DirectionsWithCounts with all Directions
-            var statistics = allDirections
-                .GroupJoin(
-                    directionsWithCounts,
-                    direction => direction.Id,
-                    directionWithCounts => directionWithCounts.DirectionId,
-                    (direction, localDirectionsWithCounts) => new { direction, localDirectionsWithCounts })
-                .SelectMany(
-                    x => x.localDirectionsWithCounts,
-                    (x, y) => new DirectionStatistic
+
+                var directionsWithApplications = applications
+                    .GroupBy(a => a.Workshop.DirectionId)
+                    .Select(g => new
                     {
-                        Direction = x.direction.ToModel(),
-                        ApplicationsCount = y.ApplicationsCount ?? 0,
-                        WorkshopsCount = y.WorkshopsCount ?? 0,
+                        DirectionId = g.Key,
+                        ApplicationsCount = g.Count() as int?,
                     });
 
-            var sortedStatistics = await statistics
-                .OrderByDescending(s => s.ApplicationsCount)
-                .Take(limit)
-                .ToListAsync()
-                .ConfigureAwait(false);
-
-            logger.LogInformation($"All {sortedStatistics.Count} records were successfully received");
-
-            return sortedStatistics;
+                // LEFT JOIN DirectionsWithWorkshops with DirectionsWithApplications
+                var directionsWithCounts = directionsWithWorkshops
+                    .GroupJoin(
```

Comment:
Probably you don't need `GroupJoin()` here, and `Join()` [Join doc](https://docs.microsoft.com/en-us/dotnet/api/system.linq.queryable.join?view=net-6.0#system-linq-queryable-join-4(system-linq-iqueryable((-0))-system-collections-generic-ienumerable((-1))-system-linq-expressions-expression((system-func((-0-2))))-system-linq-expressions-expression((system-func((-1-2))))-system-linq-expressions-expression((system-func((-0-1-3)))))) is better?